CMakeLists.txt

```diff
@@ -106,4 +111,24 @@ if(ENABLE_TEST)
     add_subdirectory(test/)
 endif()
 
+# documentation
+if(BUILD_DOCUMENTATION)
+    if(NOT DOXYGEN_FOUND)
+        message(FATAL_ERROR "Doxygen is needed to build the documentation.")
+    endif()
+
+    set(doxyfile_in ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in)
+    set(doxyfile ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
+
+    configure_file(${doxyfile_in} ${doxyfile} @ONLY)
+
+    add_custom_target(doc ALL
```

Comment:
I don't know, it's just that at the end we also have a directory `latex` and not just the `html` one